R/csv.R

```diff
@@ -100,6 +119,55 @@ duckdb_read_csv <- function(
   invisible(out)
 }
 
+
+#' Column names and types logic for duckdb_read_csv()
+#'
+#' @param found the detected (found) header and types from `utils::read_csv`
+#' @param col.names user provided column names
+#' @param col.types user provider column types and maybe names too
+#'
+#' @return returns a valid fields argument for `dbCreateTable`
+set_csv_fields <- function(found, col.names, col.types) {
+  if (is.null(col.types) && is.null(col.types)) {
+    return(found)
+  }
+
+  if (!is.null(names(col.types)) && !is.null(col.names)) {
+    warning("Ignoring `col.names` as column names provided by `col.types` parameter")
+    return(col.types)
+  }
+
+  if (!is.null(col.types)) {
+    if (length(col.types) != ncol(found)) {
+      stop(
+        "You supplied ", length(col.types), " values to `col.names`, but file has ",
+        ncol(found), " columns."
+      )
+    }
+
+    if (!is.null(names(col.types))) {
+      return(col.types)
+    } else {
+      if (length(col.types) != ncol(found)) {
+        stop(
+          "You supplied ", length(col.types), " values to `col.types`, but file has ",
+          ncol(found), " columns."
+        )
+      }
+      fields <- col.types
+      name(fields) <- col.names
```

Comment:
`R CMD check` complains, maybe:

```suggestion
      names(fields) <- col.names
```